Mu2eG4/src/Mu2eG4UserHelpers.cc

```diff
@@ -64,15 +64,28 @@ namespace mu2e {
 
     }
 
+    // The following functions attempt to get the particle end
+    // parameters. For the one which have a chance to travel, the
+    // Mu2eRecorderProcess stores the information in the
+    // Mu2eG4UserTrackInformation just before they interact
+    // inelastically or decay. For the particles born at rest, such
+    // information is not provided and their times are taken from the
+    // pre step point. Note that for the particles decaying in flight,
+    // the end times are the ones after they decay, while for the ones
+    // decaying at rest, the recorded times are the ones before they
+    // decay
+
     // global time at the point of interaction/decay
     double getEndGlobalTime(G4Track const* const trk) {
       auto const* uti = dynamic_cast<Mu2eG4UserTrackInformation*>(trk->GetUserInformation());
-      return (uti->GetGlobalTime() >= 0.) ? uti->GetGlobalTime() : trk->GetGlobalTime();
```

Comment:
I would like to make sure that I understand how this interacts with resampling stopped particles.  In particular how it interacts with PR #810.  In the job that creates stopped particles, uti->GetGlobalTime is always  >=0. because the particle has taken some steps to get to the point where it stopped and the recorder process has been run at least once.  In the resampling job if we resample a stopped particle, and it decays on it's first step, then the recorder process will not have been run and GetGlobalTIme will return -1.  Is that correct?

Reply:
Yes. For the particles created at rest, only the at rest process(es) are tried/invoked, 
therefore the recorder process which is an along process is not invoked.
The Mu2eG4UserTrackInformation initializes both times to -1. at creation.
Without this PR the time after the decay is stored in SimParticle is such cases.
I tested this with g4study, but it would be good if someone else than me also tested it.

Reply:
I am not aware that we have any working code to test this.  We will when PR #810 is updated to reflect the recent changes.  I think that is the right time to test it:

@brownd1978  Do you agree that testing this PR needs to be integrated into testing #810 or do you see another way to test?

Reply:
Doesn't this code affect all stops?   If so, validation should test this through the pileup job, which uses stopped muons.

Reply:
> Doesn't this code affect all stops? If so, validation should test this through the pileup job, which uses stopped muons.

Yes, it does. It actually runs for all the tracks.

Reply:
> Doesn't this code affect all stops? If so, validation should test this through the pileup job, which uses stopped muons.

I believe that the nightly ceMix does pileup which will test the mu- stops.   So I merged it to let the nightly do the test.  Now that it is merged Shihua's use case will be covered when he updates his PR and tests.